enh(epp): paloalto standard snmp (#2415)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/network-firewalls-paloalto-standard-snmp.md

| WARNINGOUTVOLUME       |                                                                                                                                                                                          |                                                      |             |
| CRITICALOUTVOLUME      |                                                                                                                                                                                          |                                                      |             |
| WARNINGSPEED           | Thresholds                                                                                                                                                                               |                                                      |             |
| CRITICALSPEED          | Thresholds                                                                                                                                                                               |                                                      |             |

> | CRITICALSTATUS         | Set critical threshold for status (Default: '%{admstatus} eq "up" and %{opstatus} ne "up"'). You can use the following variables: %{admstatus}, %{opstatus}, %{duplexstatus}, %{display} | %{admstatus} eq "up" and %{opstatus} !~ /up|dormant/ |             |

Reviewer: for status

> | WARNINGSTATUS          | Set warning threshold for status. You can use the following variables: %{admstatus}, %{opstatus}, %{duplexstatus}, %{display}                                                            |                                                      |             |

Reviewer: for status

|:--------------|:----------------------------------------------------------------------------------------------------|:------------------|:------------|
| WARNINGUSAGE  | Threshold warning                                                                                   |                   |             |
| CRITICALUSAGE | Threshold critical                                                                                  |                   |             |
| EXTRAOPTIONS  | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) | --verbose         |             |

</TabItem>
<TabItem value="Panorama" label="Panorama">

| Macro          | Description                                                                                                                              | Default value                 | Mandatory   |
|:---------------|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------------------------|:------------|

> | CRITICALSTATUS | Set critical threshold for status (Default: '%{status} =~ /not-connected/i'). You can use the following variables: %{status}, %{display} | %{status} =~ /not-connected/i |             |

Reviewer: for status

> | WARNINGSTATUS  | Set warning threshold for status (Default: ''). You can use the following variables: %{status}, %{display}                               |                               |             |

Reviewer: for status

| --change-exit                              | Replace an exit code with one of your choice. Eg: adding --change-exit=unknown=critical will result in a CRITICAL state instead of an UNKNOWN state.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | Output |

> | --range-perfdata                           | Change perfdata range thresholds display: 1 = start value equals to '0' is removed, 2 = threshold range is not display.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | Output |

Reviewer: ```suggestion
| --range-perfdata                           | Change perfdata range thresholds display: 1 = start value equals to '0' is removed, 2 = threshold range is not displayed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | Output |
```

> | --filter-uom                               | Masks the units when they don't match the given regular expression.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | Output |

Reviewer: ```suggestion
| --filter-uom                               | Mask the units when they don't match the given regular expression.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | Output |
```

| --maxrepetitions                           | Max repetitions value (default: 50) (only for SNMP v2 and v3).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | SNMP   |

> | --subsetleef                               | How many oid values per SNMP request (default: 50) (for get\_leef method. Be cautious when you set it. Prefer to let the default value).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | SNMP   |

Reviewer: ```suggestion
| --subsetleef                               | How many OID values per SNMP request (default: 50) (for get\_leef method. Be cautious when you set it. Prefer to let the default value).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | SNMP   |
```

> | --snmp-autoreduce                          | Progressively reduce the number requested OIDs in bulk mode. Use it in case of SNMP errors (By default, the number is divided by 2).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | SNMP   |

Reviewer: ```suggestion
| --snmp-autoreduce                          | Progressively reduce the number of requested OIDs in bulk mode. Use it in case of SNMP errors (By default, the number is divided by 2).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | SNMP   |
```

> | --snmp-force-getnext                       | Use snmp getnext function in snmp v2c and v3. This will request one OID at a time.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | SNMP   |

Reviewer: ```suggestion
| --snmp-force-getnext                       | Use SNMP getnext function in snmp v2c and v3. This will request one OID at a time.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | SNMP   |
```

</TabItem>
<TabItem value="Global-Protect-Tunnels" label="Global-Protect-Tunnels">

| Option                   | Description                                                                             | Type |
|:-------------------------|:----------------------------------------------------------------------------------------|:-----|
| --warning-* --critical-* | Thresholds. Can be: 'tunnels-usage', 'tunnels-usage-free', 'tunnels-usage-prct' (%).    | Mode |

</TabItem>
<TabItem value="Hardware" label="Hardware">

| Option               | Description                                                                                                                                                                                      | Type |
|:---------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----|
| --component          | Which component to check (Default: '.*'). Can be: 'device'.                                                                                                                                      | Mode |

> | --filter             | Exclude some parts (comma seperated list) Can also exclude specific instance: --filter=device,network.*                                                                                          | Mode |

Reviewer: either "can also exclude a specific instance" or "can also exclude specific instances"

> | --no-component       | Return an error if no compenents are checked. If total (with skipped) is 0. (Default: 'critical' returns).                                                                                       | Mode |

Reviewer: ?

> | --threshold-overload | Set to overload default threshold values (syntax: section,\[instance,\]status,regexp) It used before default thresholds (order stays). Example: --threshold-overload='device.network,OK,down'    | Mode |

Reviewer: ```suggestion
| --threshold-overload | Use this option to overload the default threshold values (syntax: section,\[instance,\]status,regexp). It can be used before default thresholds (order stays). Example: --threshold-overload='device.network,OK,down'    | Mode |
```

| --add-speed              | Check interface speed.                                                                                                                                                                                                                                                                     | Mode      |
| --add-volume             | Check interface data volume between two checks (not supposed to be graphed, useful for BI reporting).                                                                                                                                                                                      | Mode      |
| --check-metrics          | If the expression is true, metrics are checked (Default: '%{opstatus} eq "up"').                                                                                                                                                                                                           | Mode      |

> | --warning-status         | Set warning threshold for status. You can use the following variables: %{admstatus}, %{opstatus}, %{duplexstatus}, %{display}                                                                                                                                                              | Mode      |

Reviewer: for status

| --units-cast             | Units of thresholds for communication types (Default: 'percent\_delta') ('percent\_delta', 'percent', 'delta', 'deltaps', 'counter').                                                                                                                                                      | Mode      |
| --nagvis-perfdata        | Display traffic perfdata to be compatible with nagvis widget.                                                                                                                                                                                                                              | Mode      |
| --interface              | Set the interface (number expected) ex: 1,2,... (empty means 'check all interface').                                                                                                                                                                                                       | Mode      |

> | --name                   | Allows to use interface name with option --interface instead ofinterface oid index (Can be a regexp)                                                                                                                                                                                       | Mode      |

Reviewer: ```suggestion
| --name                   | Allows to use interface name with option --interface instead ofinterface OID index (can be a regexp)                                                                                                                                                                                       | Mode      |
```

| --show-cache           | Display cache storage datas.                                                                                                                                                                                                                  | Mode      |
| --filter-storage-type  | Filter storage types with a regexp (Default: '^(hrStorageRam\|hrStorageFlashMemory)$').                                                                                                                                                       | Mode      |

</TabItem>
<TabItem value="Panorama" label="Panorama">

| Option            | Description                                                                                                                                 | Type |
|:------------------|:--------------------------------------------------------------------------------------------------------------------------------------------|:-----|

> | --warning-status  | Set warning threshold for status (Default: ''). You can use the following variables: %{status}, %{display}                                  | Mode |

Reviewer: for status?

> | --critical-status | Set critical threshold for status (Default: '%{status} =~ /not-connected/i'). You can use the following variables: %{status}, %{display}    | Mode |

Reviewer: for status
